crates/biome_html_analyze/src/lint/a11y/use_iframe_title.rs

```diff
@@ -0,0 +1,163 @@
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_diagnostics::Severity;
+use biome_html_syntax::AnyHtmlElement;
+use biome_rowan::{AstNode, TextRange};
+use biome_rule_options::use_iframe_title::UseIframeTitleOptions;
+
+declare_lint_rule! {
+    /// Enforces the usage of the attribute `title` for the element `iframe`.
+    ///
+    /// :::note
+    /// In `.html` files, this rule matches `iframe` elements case-insensitively (e.g., `<IFRAME>`, `<IFrame>`).
+    ///
+    /// In component-based frameworks (Vue, Svelte, Astro), only lowercase `<iframe>` is checked. PascalCase variants like `<Iframe>` are assumed to be custom components and are ignored.
+    /// :::
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```html,expect_diagnostic
+    /// <iframe></iframe>
+    /// ```
+    ///
+    /// ```html,expect_diagnostic
+    /// <iframe title=""></iframe>
+    /// ```
+    ///
+    /// ```html,expect_diagnostic
+    /// <IFRAME></IFRAME>
+    /// ```
+    ///
+    /// ```vue,expect_diagnostic
+    /// <iframe></iframe>
+    /// ```
+    ///
+    /// ```vue,expect_diagnostic
+    /// <iframe title=""></iframe>
+    /// ```
+    ///
+    /// ```svelte,expect_diagnostic
+    /// <iframe></iframe>
+    /// ```
+    ///
+    /// ```svelte,expect_diagnostic
+    /// <iframe title=""></iframe>
+    /// ```
+    ///
+    /// ```astro,expect_diagnostic
+    /// <iframe></iframe>
+    /// ```
+    ///
+    /// ```astro,expect_diagnostic
+    /// <iframe title=""></iframe>
+    /// ```
```

Comment:
Same here, tbh. Too many examples. We should demonstrate that when `title`​ is missing its invalid, and when `title`​ is empty its invalid.

While the documentation _is_ tested, its not unit tests. Too much information can harm clarity.